test/e2e/queue.go

```diff
@@ -19,14 +19,187 @@ package e2e
+					Expect(err).NotTo(HaveOccurred(), "Get queue %s failed", q1)
+					return queue.Status.Running > 0, nil
+				})
+				Expect(err).NotTo(HaveOccurred(), "Error waiting for queue running")
+			})
+		})
+
+		Context("When all of pod deleted and lackof resources pod group status should transform from Running to Pending", func() {
+			It("Transform from running to pending should succeed", func() {
+				By("Verify queue have pod groups running")
+				err := waitQueueStatus(func() (bool, error) {
+					queue, err := ctx.vcclient.SchedulingV1beta1().Queues().Get(context.TODO(), q1, metav1.GetOptions{})
+					Expect(err).NotTo(HaveOccurred(), "Get queue %s failed", q1)
+					return queue.Status.Running > 0, nil
+				})
+				Expect(err).NotTo(HaveOccurred(), "Error waiting for queue running")
+
+				clusterPods, err := ctx.kubeclient.CoreV1().Pods(podNamespace).List(context.TODO(), metav1.ListOptions{})
+				for _, pod := range clusterPods.Items {
+					if pod.Labels["volcano.sh/job-name"] == firstJobName {
+						err = ctx.kubeclient.CoreV1().Pods(podNamespace).Delete(context.TODO(), pod.Name, metav1.DeleteOptions{})
+						Expect(err).NotTo(HaveOccurred(), "Failed to delete pod %s", pod.Name)
+					}
+				}
+
+				By("Verify queue have pod groups Pending")
+				err = waitQueueStatus(func() (bool, error) {
+					queue, err := ctx.vcclient.SchedulingV1beta1().Queues().Get(context.TODO(), q1, metav1.GetOptions{})
+					Expect(err).NotTo(HaveOccurred(), "Get queue %s failed", q1)
+					return queue.Status.Pending > 0, nil
+				})
+				Expect(err).NotTo(HaveOccurred(), "Error waiting for queue Pending")
+			})
+		})
+
+		Context("When some of pod deleted and lackof resources pod group status should transform from Running to Unknown", func() {
+			It("Transform from running to unknown should succeed", func() {
+				By("Verify queue have pod groups running")
+				err := waitQueueStatus(func() (bool, error) {
+					queue, err := ctx.vcclient.SchedulingV1beta1().Queues().Get(context.TODO(), q1, metav1.GetOptions{})
+					Expect(err).NotTo(HaveOccurred(), "Get queue %s failed", q1)
+					return queue.Status.Running > 0, nil
+				})
+				Expect(err).NotTo(HaveOccurred(), "Error waiting for queue running")
+
+				By("Delete some of pod which will case pod group status transform from running to unknown.")
+				podDeleteNum := 0
+
+				err = waitPodPhaseRunningMoreThanNum(ctx, podNamespace, 2)
+				Expect(err).NotTo(HaveOccurred(), "Failed waiting for pods")
+
+				clusterPods, err := ctx.kubeclient.CoreV1().Pods(podNamespace).List(context.TODO(), metav1.ListOptions{})
+				for _, pod := range clusterPods.Items {
+					if pod.Status.Phase == corev1.PodRunning {
+						err = ctx.kubeclient.CoreV1().Pods(podNamespace).Delete(context.TODO(), pod.Name, metav1.DeleteOptions{})
+						Expect(err).NotTo(HaveOccurred(), "Failed to delete pod %s", pod.Name)
+						podDeleteNum = podDeleteNum + 1
+					}
+					if podDeleteNum >= int(rep/2) {
```

Comment:
I can hardly understand what these above codes are doing

Reply:
This cases is not easy to reappear， i need to make sure there is some of pod is in running state, and locate then and delete (restart) them which will case an unknown state.